event-handlers/src/main/java/no/sikt/nva/nvi/events/cristin/CristinMapper.java

```diff
@@ -236,21 +238,51 @@ private static boolean hasInstitutionPoints(ScientificPerson scientificPerson) {
         return nonNull(scientificPerson.getAuthorPointsForAffiliation());
     }
 
-    private static DbInstitutionPoints toDbInstitutionPoints(Entry<URI, BigDecimal> entry) {
-        //TODO: Implement creatorAffiliationPoints
-        return new DbInstitutionPoints(entry.getKey(), entry.getValue(), List.of());
+    private static DbInstitutionPoints toDbInstitutionPoints(InstitutionPoints institutionPoints) {
+        return new DbInstitutionPoints(institutionPoints.institutionId(), institutionPoints.points(),
+                                       toCreatorPoints(institutionPoints));
+    }
+
+    private static List<DbCreatorAffiliationPoints> toCreatorPoints(InstitutionPoints institutionPoints) {
+        return institutionPoints.creatorPoints()
+                   .stream()
+                   .map(CristinMapper::getDbCreatorAffiliationPoints)
+                   .collect(Collectors.toList());
+    }
+
+    private static DbCreatorAffiliationPoints getDbCreatorAffiliationPoints(CreatorPoints creatorPoints) {
+        return new DbCreatorAffiliationPoints(creatorPoints.creatorId(), creatorPoints.affiliationId(),
+                                              creatorPoints.points());
     }
 
     private static List<ScientificPerson> getCreators(CristinNviReport cristinNviReport) {
         return cristinNviReport.scientificResources().get(0).getCreators();
     }
 
-    private static Collector<ScientificPerson, ?, Map<URI, BigDecimal>> collectToMapOfPoints(
+    private static Collector<ScientificPerson, ?, List<InstitutionPoints>> collectToInstitutionOfPoints(
         List<CristinLocale> institutions) {
-        return Collectors.groupingBy(scientificPerson -> getTopLevelOrganization(scientificPerson, institutions),
-                                     Collectors.reducing(BigDecimal.ZERO.setScale(4, RoundingMode.HALF_UP),
-                                                         CristinMapper::extractAuthorPointsForAffiliation,
-                                                         BigDecimal::add));
+        return Collectors.collectingAndThen(
+            Collectors.groupingBy(scientificPerson -> getTopLevelOrganization(scientificPerson, institutions),
+                                  Collectors.toCollection(ArrayList::new)), map -> map.entrySet()
+                                                                                       .stream()
+                                                                                       .map(entry -> toPoints(
+                                                                                           institutions,
+                                                                                           entry.getValue()))
+                                                                                       .collect(Collectors.toList()));
```

Comment:
this part is new, for every institution, iterating throw creators, if institution id match (first number in cristinId) add creator to affiliation points for matching institution.